core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -385,21 +387,18 @@ func (eb *EthBroadcaster) processUnstartedEthTxs(ctx context.Context, fromAddres
 		n++
 		var a EthTxAttempt
 		keySpecificMaxGasPriceWei := eb.config.KeySpecificMaxGasPriceWei(etx.FromAddress)
+		fee, gasLimit, err := eb.estimator.GetFee(ctx, etx.EncodedPayload, etx.GasLimit, keySpecificMaxGasPriceWei)
+		if err != nil {
+			return errors.Wrap(err, "failed to get fee"), true
+		}
+
 		if eb.config.EvmEIP1559DynamicFees() {
-			fee, gasLimit, err := eb.estimator.GetDynamicFee(ctx, etx.GasLimit, keySpecificMaxGasPriceWei)
-			if err != nil {
-				return errors.Wrap(err, "failed to get dynamic gas fee"), true
-			}
-			a, err = eb.NewDynamicFeeAttempt(*etx, fee, gasLimit)
+			a, err = eb.NewDynamicFeeAttempt(*etx, *fee.Dynamic, gasLimit)
```

Comment:
I assume the legacy/dynamic details still inside the TxMgr are because you plan to tackle this later while designing the TxAttemptBuilder right?

Reply:
yes that's correct! the `TxAttemptBuilder` will contain the rest of the building logic once it's extracted